### PR TITLE
Update usage.md

### PR DESCRIPTION
**Pull request by ojziff**
easier to read and follow

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
  - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/rnafusion/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/rnafusion _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by ojziff** [comment]
make usage docs easier to read and make easier to use for the novice

File: docs/usage.md [hunk 1]
```diff
@@ -8,100 +8,83 @@
...
-   - Fastqc
-   - MultiQC
-   - arriba visualisation (for fusion detected by arriba only)
-   - fusion-report
-   - fusionInspector
-
-### Prerequisite: download and build references
+1. Download and build references 
+   - specified with `--build_references` parameter
+   - this is required only once before running the pipeline.
```
Comment (matq007) [inline docs/usage.md:13]:
```suggestion
  - required only once before running the pipeline
```

File: docs/usage.md [hunk 2]
```diff
@@ -8,100 +8,83 @@
...
-   - MultiQC
-   - arriba visualisation (for fusion detected by arriba only)
-   - fusion-report
-   - fusionInspector
-
-### Prerequisite: download and build references
+1. Download and build references 
+   - specified with `--build_references` parameter
+   - this is required only once before running the pipeline. 
+   - It should be rerun after pipeline updates.
```
Comment (matq007) [inline docs/usage.md:14]:
```suggestion
  - **Important**: rerun with each new release
```

File: docs/usage.md [hunk 3]
```diff
@@ -8,100 +8,83 @@
...
+2. Detecting fusions 
+   - This uses any combination of the fusion tools: arriba, fusioncatcher, pizzly, squid, starfusion.
```
Comment (matq007) [inline docs/usage.md:16]:
```suggestion
  - Supported tools: `arriba`, `fusioncatcher`, `pizzly`, `squid` and `starfusion`
```

File: docs/usage.md [hunk 4]
```diff
@@ -8,100 +8,83 @@
...
+   - QC is performed with: Fastqc and MultiQC.
```
Comment (matq007) [inline docs/usage.md:17]:
```suggestion
  - QC: `Fastqc` and `MultiQC`
```

File: docs/usage.md [hunk 5]
```diff
@@ -8,100 +8,83 @@
...
+   - Visualisation of fusions is performed with: arriba visualisation (for fusion detected by arriba only), fusion-report, fusionInspector
```
Comment (matq007) [inline docs/usage.md:18]:
```suggestion
  - Fusion visualization: `arriba` (only arriba detected), `fusion-report` and `fusionInspector`
```

File: docs/usage.md [hunk 6]
```diff
@@ -8,100 +8,83 @@
...
+
+### 1. Download and build references
 
 The rnafusion pipeline needs references for the fusion detection tools, so downloading these is a **requirement**.
-It is possible to download and build each reference manually (for example in case of non-human samples that are not supported currently) feed references manually to rnafusion using the arguments `--<tool>_ref` but it is advised to download references with rnafusion:
+Whilst it is possible to download and build each reference manually, it is advised to download references with the rnafusion pipeline.
+
+First register for a free account at COSMIC at https://cancer.sanger.ac.uk/cosmic/register using your university email ending .ac.uk. The account is only activated upon clicking the link in the registation email.
```
Comment (matq007) [inline docs/usage.md:25]:
```suggestion
First register for a free account at COSMIC at https://cancer.sanger.ac.uk/cosmic/register using your university email. The account is **only activated upon** clicking the link in the registration email.
```

File: docs/usage.md [hunk 7]
```diff
@@ -8,100 +8,83 @@
...
+
+Then run the pipeline, ensuring to include in the command line your COSMIC registration email and password as well as the path to the desired reference output directory:
```
Comment (matq007) [inline docs/usage.md:27]:
```suggestion
Download the references as shown below including your COSMIC credentials. Note that this step can take up to 24 hours to complete.
```

File: docs/usage.md [hunk 8]
```diff
@@ -8,100 +8,83 @@
...
 
 ```bash
 nextflow run nf-core/rnafusion \
 --build_references --all \
---genomes_base <PATH> \
+--cosmic_username <EMAIL> --cosmic_passwd <PASSWORD> \
+--outdir <PATH> -r 2.0.0
```
Comment (matq007) [inline docs/usage.md:33]:
```suggestion
nextflow run nf-core/rnafusion \
  --build_references --all \
  --cosmic_username <EMAIL> --cosmic_passwd <PASSWORD> \
  --outdir <PATH>
```

File: docs/usage.md [hunk 9]
```diff
@@ -8,100 +8,83 @@
...
 ```
 
-References for the different tools can also be downloaded separately:
+This step can take over 24 hours to complete.
+
```
Comment (matq007) [inline docs/usage.md:37]:
```suggestion
```

File: docs/usage.md [hunk 10]
```diff
@@ -8,100 +8,83 @@
...
+References for each tools can also be downloaded separately with:
 
 ```bash
 nextflow run nf-core/rnafusion \
 --build_references --<tool> \
---genomes_base <PATH>
+--cosmic_username <EMAIL> --cosmic_passwd <PASSWORD> \
+--outdir <PATH> -r 2.0.0
```
Comment (matq007) [inline docs/usage.md:44]:
```suggestion
nextflow run nf-core/rnafusion \
  --build_references --<tool> \
  --cosmic_username <EMAIL> --cosmic_passwd <PASSWORD> \
  --outdir <PATH>
```

File: docs/usage.md [hunk 11]
```diff
@@ -8,100 +8,83 @@
...
+In the case of non-human references, that are not supported currently, these can be fed manually to rnafusion using the parameter `--<tool>_ref`. By default STAR-Fusion references are built. You can also download them from [CTAT](https://github.com/NCIP/Trinity_CTAT/wiki). This allows more flexibility for different organisms but be aware that this is not fully tested:
 
 ```bash
 nextflow run nf-core/rnafusion \
 --build_references --starfusion/--all \
---starfusion_build false \
---genomes_base <PATH>
+--starfusion_build FALSE \
+--cosmic_username <EMAIL> --cosmic_passwd <PASSWORD> \
+--outdir <PATH> -r 2.0.0
```
Comment (matq007) [inline docs/usage.md:54]:
```suggestion
nextflow run nf-core/rnafusion \
  --build_references --all \
  --starfusion_build FALSE \
  --cosmic_username <EMAIL> --cosmic_passwd <PASSWORD> \
  --outdir <REFERENCE_PATH>
```

File: docs/usage.md [hunk 12]
```diff
@@ -8,100 +8,83 @@
...
 ```
 
 Then use the flag `--starfusion_build` while running the detection.
 
-### Running all detection tools
+### 2. Detecting fusions
+
+This step can either be run using all fusion detection tools or specifying individual tools. Visualisation tools will be run on all fusions detected. To run all tools (arriba, fusioncatcher, pizzly, squid, starfusion) use the `--all` parameter:
```
Comment (matq007) [inline docs/usage.md:61]:
```suggestion
This step can either be run using all fusion detection tools or specifying individual tools. Visualisation tools will be run on all fusions detected. To run all tools (`arriba`, `fusioncatcher`, `pizzly`, `squid`, `starfusion`) use the `--all` parameter:
```

File: docs/usage.md [hunk 13]
```diff
@@ -8,100 +8,83 @@
...
-
-### Running a specific detection tool
+Alternatively, to run only a specific detection tool specify with `--tool`:
 
 ```bash
 nextflow run nf-core/rnafusion \
---input '[path to samplesheet file]' --<tool> \
+--<tool> \
+--input '[path to samplesheet file]' \
 --outdir <PATH> \
```
Comment (matq007) [inline docs/usage.md:78]:
```suggestion
nextflow run nf-core/rnafusion \
  --<tool> \
  --input <SAMPLE_SHEET.CSV> \
  --outdir <PATH> \
  --genomes_base <REFERENCE_PATH>
```

File: docs/usage.md [hunk 14]
```diff
@@ -8,100 +8,83 @@
...
+This step can either be run using all fusion detection tools or specifying individual tools. Visualisation tools will be run on all fusions detected. To run all tools (arriba, fusioncatcher, pizzly, squid, starfusion) use the `--all` parameter:
 
 ```bash
 nextflow run nf-core/rnafusion \
---input '[path to samplesheet file]' --all \
+--all \
+--input '[path to samplesheet csv]' \
 --outdir <PATH> \
+--genomes_base 'path to rnafusion references' \
+-r 2.0.0
```
Comment (matq007) [inline docs/usage.md:69]:
```suggestion
nextflow run nf-core/rnafusion \
  --all \
  --input <SAMPLE_SHEET.CSV> \
  --outdir <PATH> \
  --genomes_base <REFERENCE_PATH>
```

File: docs/usage.md [hunk 15]
```diff
@@ -8,100 +8,83 @@
...
-CONTROL_REP1,AEG588A1_S1_L004_R1_001.fastq.gz,AEG588A1_S1_L004_R2_001.fastq.gz
-```
-
-### Full samplesheet
-
-The pipeline will auto-detect whether a sample is single- or paired-end using the information provided in the samplesheet. The samplesheet can have as many columns as you desire, however, there is a strict requirement for the first 4 columns to match those defined in the table below.
-
-A final samplesheet file consisting of both single- and paired-end data may look something like the one below. This is for 6 samples, where `TREATMENT_REP3` has been sequenced twice.
+You will need to create a samplesheet with information about the samples you would like to analyse before running the pipeline. Use the `--input` parameter to specify its location. The pipeline will detect whether a sample is single- or paired-end from the samplesheet - the fastq_2 column is empty for single-end. The samplesheet has to be a comma-separated file (.csv) but can have as many columns as you desire. There is a strict requirement for the first 4 columns to match those defined in the table below with the header row included.
```
Comment (matq007) [inline docs/usage.md:87]:
```suggestion
You will need to create a samplesheet with information about the samples you would like to analyse before running the pipeline. Use the `--input` parameter to specify its location. The pipeline will detect whether a sample is single- or paired-end from the samplesheet - the `fastq_2` column is empty for single-end. The samplesheet has to be a comma-separated file (.csv) but can have as many columns as you desire. There is a strict requirement for the first 4 columns to match those defined in the table below with the header row included.
```

**Review by matq007 — COMMENTED** [review]
Some suggestions

**Comment by matq007** [comment]
Hey @ojziff, what about the suggestions regarding skipping version tag. We would have to update it with every new release in the docs, sounds like an extra work. Also could you also run prettier so we pass the tests? Thx

**Comment by ojziff** [comment]
I tried accepting those version tag suggestions but it wont let me with error:
```Applying suggestions on deleted lines is currently not supported.```

**Review by matq007 — APPROVED** [review]

File: docs/usage.md (hunk 8 above)
Comment (rannick) [reply docs/usage.md]: `--genomes_base` is needed to know where to write the references

File: docs/usage.md [hunk 16]
```diff
@@ -8,100 +8,81 @@
...
-It is possible to download and build each reference manually (for example in case of non-human samples that are not supported currently) feed references manually to rnafusion using the arguments `--<tool>_ref` but it is advised to download references with rnafusion:
+Whilst it is possible to download and build each reference manually, it is advised to download references with the rnafusion pipeline.
+
+First register for a free account at COSMIC at https://cancer.sanger.ac.uk/cosmic/register using your university email. The account is **only activated upon** clicking the link in the registration email.
+
+Download the references as shown below including your COSMIC credentials. Note that this step can take up to 24 hours to complete.
 
 ```bash
 nextflow run nf-core/rnafusion \
 --build_references --all \
```
Comment (rannick) [inline docs/usage.md]:
```suggestion
```bash
nextflow run nf-core/rnafusion \
  --build_references --all \
  --genomes_base <PATH> \
```

File: docs/usage.md [hunk 17]
```diff
@@ -8,100 +8,81 @@
...
---genomes_base <PATH> \
+--cosmic_username <EMAIL> --cosmic_passwd <PASSWORD> \
+--outdir <PATH> -r 2.0.0
```
Comment (rannick) [inline docs/usage.md]: Line to update with indentation and removing -r 2.0.0

File: docs/usage.md [hunk 18]
```diff
@@ -8,100 +8,81 @@
...
 --build_references --<tool> \
---genomes_base <PATH>
+--cosmic_username <EMAIL> --cosmic_passwd <PASSWORD> \
+--outdir <PATH> -r 2.0.0
 ```
 
-This PATH will be the place the references will be saved.
-
-Optional: by default STAR-Fusion references are built. You can also download them from CTAT. This allows more flexibility for different organisms but be aware that **this is not fully tested -> not recommended**:
+In the case of non-human references, that are not supported currently, these can be fed manually to rnafusion using the parameter `--<tool>_ref`. By default STAR-Fusion references are built. You can also download them from [CTAT](https://github.com/NCIP/Trinity_CTAT/wiki). This allows more flexibility for different organisms but be aware that this is not fully tested:
```
Comment (rannick) [inline docs/usage.md:45]:
```suggestion
**Optional:**
In the case of non-human references, that are not supported currently, these can be fed manually to rnafusion using the parameter `--<tool>_ref`. By default STAR-Fusion references are built. rnafusion also has an option to download the references from [CTAT](https://github.com/NCIP/Trinity_CTAT/wiki). This allows more flexibility for different organisms but be aware that **this is not fully tested and therefore not recommended**
```

File: docs/usage.md [hunk 19]
```diff
@@ -8,100 +8,81 @@
...
+--input '[path to samplesheet csv]' \
 --outdir <PATH> \
+--genomes_base 'path to rnafusion references' \
+-r 2.0.0
 ```
 
-Visualisation tools will be run on all fusions detected.
-
-### Running a specific detection tool
+Alternatively, to run only a specific detection tool specify with `--tool`:
```
Comment (rannick) [inline docs/usage.md:69]:
```suggestion
Alternatively, to run only a specific detection tool specified with `--<tool>`:
```

File: docs/usage.md [hunk 20]
```diff
@@ -8,100 +8,81 @@
...
-CONTROL_REP1,AEG588A1_S1_L004_R1_001.fastq.gz,AEG588A1_S1_L004_R2_001.fastq.gz
-```
-
-### Full samplesheet
-
-The pipeline will auto-detect whether a sample is single- or paired-end using the information provided in the samplesheet. The samplesheet can have as many columns as you desire, however, there is a strict requirement for the first 4 columns to match those defined in the table below.
-
-A final samplesheet file consisting of both single- and paired-end data may look something like the one below. This is for 6 samples, where `TREATMENT_REP3` has been sequenced twice.
+You will need to create a samplesheet with information about the samples you would like to analyse before running the pipeline. Use the `--input` parameter to specify its location. The pipeline will detect whether a sample is single- or paired-end from the samplesheet - the `fastq_2` column is empty for single-end. The samplesheet has to be a comma-separated file (.csv) but can have as many columns as you desire. There is a strict requirement for the first 4 columns to match those defined in the table below with the header row included.
```
Comment (rannick) [inline docs/usage.md:84]:
```suggestion
You will need to create a samplesheet with information about the samples you would like to analyse before running the pipeline. Use the `--input` parameter to specify its location. It has to be a comma-separated file with 4 columns, and a header row as shown in the examples below.

```console
--input '[path to samplesheet file]'
```

### Multiple runs of the same sample

The `sample` identifiers have to be the same when you have re-sequenced the same sample more than once e.g. to increase sequencing depth. The pipeline will concatenate the raw reads before performing any downstream analysis. Below is an example for the same sample sequenced across 3 lanes:

```console
sample,fastq_1,fastq_2,strandedness
CONTROL_REP1,AEG588A1_S1_L002_R1_001.fastq.gz,AEG588A1_S1_L002_R2_001.fastq.gz,forward
CONTROL_REP1,AEG588A1_S1_L003_R1_001.fastq.gz,AEG588A1_S1_L003_R2_001.fastq.gz,forward
CONTROL_REP1,AEG588A1_S1_L004_R1_001.fastq.gz,AEG588A1_S1_L004_R2_001.fastq.gz,forward
```

### Full samplesheet

The pipeline will auto-detect whether a sample is single- or paired-end using the information provided in the samplesheet. The samplesheet can have as many columns as you desire, however, there is a strict requirement for the first 4 columns to match those defined in the table below.

A final samplesheet file consisting of both single- and paired-end data may look something like the one below. This is for 6 samples, where `TREATMENT_REP3` has been sequenced twice.
```

File: docs/usage.md [hunk 21]
```diff
@@ -114,6 +95,8 @@ TREATMENT_REP3,AEG588A6_S6_L003_R1_001.fastq.gz,,forward
 TREATMENT_REP3,AEG588A6_S6_L004_R1_001.fastq.gz,,forward
 ```
 
+A final samplesheet file consisting of both single- and paired-end data may look something like the one below. This is for 6 samples, where `TREATMENT_REP3` has been sequenced twice.
+
```
Comment (rannick) [inline docs/usage.md:98]:
```suggestion
```

File: docs/usage.md [hunk 22]
```diff
@@ -123,6 +106,15 @@ TREATMENT_REP3,AEG588A6_S6_L004_R1_001.fastq.gz,,forward
...
 
+As you can see above for multiple runs of the same sample, the `sample` name has to be the same when you have re-sequenced the same sample more than once e.g. to increase sequencing depth. The pipeline will concatenate the raw reads before performing any downstream analysis. Below is an example for the same sample sequenced across 3 lanes:
+
+```console
+sample,fastq_1,fastq_2
+CONTROL_REP1,AEG588A1_S1_L002_R1_001.fastq.gz,AEG588A1_S1_L002_R2_001.fastq.gz
+CONTROL_REP1,AEG588A1_S1_L003_R1_001.fastq.gz,AEG588A1_S1_L003_R2_001.fastq.gz
+CONTROL_REP1,AEG588A1_S1_L004_R1_001.fastq.gz,AEG588A1_S1_L004_R2_001.fastq.gz
+```
+
```
Comment (rannick) [inline docs/usage.md:116]:
```suggestion
As you can see above for multiple runs of the same sample, the `sample` name has to be the same when you have re-sequenced the same sample more than once e.g. to increase sequencing depth. The pipeline will concatenate the raw reads before performing any downstream analysis. Below is an example for the same sample sequenced across 3 lanes:

```console
sample,fastq_1,fastq_2,strandedness
CONTROL_REP1,AEG588A1_S1_L002_R1_001.fastq.gz,AEG588A1_S1_L002_R2_001.fastq.gz,forward
CONTROL_REP1,AEG588A1_S1_L003_R1_001.fastq.gz,AEG588A1_S1_L003_R2_001.fastq.gz,forward
CONTROL_REP1,AEG588A1_S1_L004_R1_001.fastq.gz,AEG588A1_S1_L004_R2_001.fastq.gz,forward
```

```

**Pull request merged**